extract_endpoint/src/extract_endpoint/azure_utils.py

```diff
@@ -17,16 +18,39 @@ class EnvVariables(enum.Enum):
     domain = 'EXTRACT_ENDPOINT_STORAGE_DOMAIN'
 
 
-def upload_bytes_to_azure(coords: StorageCoordinates, data: bytes, filename: str) -> bool:
-    azure_service = blob.BlockBlobService(account_name=coords.account,
-                                          account_key=coords.key,
-                                          custom_domain=coords.domain)
-    azure_service.create_blob_from_bytes(coords.container, filename, data)
-    return filename in [blob.name for blob in azure_service.list_blobs(coords.container)]
+class StorageProtocol(typing_extensions.Protocol):
+    def upload(self, data: bytes, filename: str) -> bool:
+        pass
 
+    def download(self, filename: str) -> bytes:
+        pass
 
-def download_bytes_from_azure(coords: StorageCoordinates, filename: str) -> str:
-    azure_service = blob.BlockBlobService(account_name=coords.account,
-                                          account_key=coords.key,
-                                          custom_domain=coords.domain)
-    return azure_service.get_blob_to_bytes(coords.container, filename).content
+
+class MockStorage:
+    def __init__(self) -> None:
+        self._data: typing.Dict[str, bytes] = {}
+
+    def upload(self, data: bytes, filename: str) -> bool:
+        self._data.update({filename: data})
+        return True
+
+    def download(self, filename: str) -> typing.Optional[bytes]:
+        if filename in self._data:
+            return self._data.get(filename)
+        else:
+            raise ValueError('File does not exist')
```

Comment:
Perhaps a `LookupError` would be a better fit than `ValueError`? I'll leave that to you if you want to change or not, not a big deal and kind of up to personal preference.

Reply:
👍